### ISSUE TITLE
cache-bundle crush symfony in dev environment

### ISSUE THREAD

**Issue by zhil** [php-cache/issues#112]
Bug description
---
Cache-bundle crush symfony in dev environment because of unserializable data in Cache\CacheBundle\DataCollector\CacheDataCollector 

Bug reproduce
---

I have added simple code based on default symfony2 https://github.com/zhil/cache-bundle-bug

Steps to prepare environment
```
git clone https://github.com/zhil/cache-bundle-bug
cd cache-bundle-bug
composer install
```

Bug can be reproduced in couple ways
```
./app/console server:run
open http://127.0.0.1:8000 in your browser - you will get exception Serialization of 'Closure' is not allowed
```
![image](https://user-images.githubusercontent.com/981783/28709799-535a9eea-738a-11e7-944c-6e3603441974.png)

Second way - use command line wrapper
```
./app/console app:test
```

You will get exception
![image](https://user-images.githubusercontent.com/981783/28709853-99f99932-738a-11e7-926b-df05cac9f255.png)

Research of the bug
---

Problem is in this code in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Profiler/FileProfilerStorage.php:158
```
if (false === file_put_contents($file, serialize($data))) {
            return false;
        }
```

$data['data']['php-cache'] - instance of Cache\CacheBundle\DataCollector\CacheDataCollector cant be serialized
![image](https://user-images.githubusercontent.com/981783/28709999-1bac3e6c-738b-11e7-9d70-d6d95bcd3d37.png)

Problem in this item $data['data']['php-cache']->getCalls()['cache.provider.apcu'][8]->argument
![image](https://user-images.githubusercontent.com/981783/28710052-4a40a51a-738b-11e7-900d-df15b1dc2db1.png)

Core problem is in @template annotation in symfony - owner is controller, it have container property, which stored as closure?

Didn't do further research

**Comment by zhil** [comment]
temporary fix for this bug - override data-collector service with in global config.
```
services:
  cache.data_collector:
    class: Cache\CacheBundle\DataCollector\CacheDataCollector
```

**Important - do not add tag data_collector**

**Comment by Nyholm** [comment]
Thank you for this issue. 
What version of the cachebundle are you using?

**Comment by zhil** [comment]
```
composer info --installed

cache/adapter-bundle            0.5.0               A bundle that registers cache implementations as Symfony services supporting PSR-6 and tagging
cache/adapter-common            0.4.0               Common classes for PSR-6 adapters
cache/apc-adapter               0.4.0               A PSR-6 cache implementation using apc. This implementation supports tags
cache/apcu-adapter              0.3.0               A PSR-6 cache implementation using apcu. This implementation supports tags
cache/cache-bundle              0.5.0               Symfony 2 bundle providing integration between PSR-6 compliant cache services and the framework. It supports cache for se...
cache/hierarchical-cache        0.4.0               A helper trait and interface to your PSR-6 cache to support hierarchical keys.
cache/memcached-adapter         0.4.0               A PSR-6 cache implementation using Memcached. This implementation supports tags
cache/namespaced-cache          0.1.3               A decorator that makes your cache support namespaces
cache/psr-6-doctrine-bridge     dev-master 541f93d  PSR-6 Doctrine bridge
cache/session-handler           0.2.1               An implementation of PHP's SessionHandlerInterface that allows PSR-6
cache/tag-interop               1.0.0               Framework interoperable interfaces for tags
cache/taggable-cache            0.5.1               Add tag support to your PSR-6 cache implementation
```

**Comment by Nyholm** [comment]
Do you have the same issue with version 1.0?

**Comment by zhil** [comment]
In test repository I have updated some of the cache libraries to
```
cache/adapter-bundle                 0.5.0              A bundle that registers cache implementations as Symfony services supporting PSR-6 and tagging
cache/adapter-common                 0.4.0              Common classes for PSR-6 adapters
cache/apc-adapter                    0.4.0              A PSR-6 cache implementation using apc. This implementation supports tags
cache/apcu-adapter                   0.3.0              A PSR-6 cache implementation using apcu. This implementation supports tags
cache/cache-bundle                   1.0.0              Symfony 2 bundle providing integration between PSR-6 compliant cache services and the framework. It supports cache fo...
cache/hierarchical-cache             0.4.0              A helper trait and interface to your PSR-6 cache to support hierarchical keys.
cache/memcached-adapter              0.4.0              A PSR-6 cache implementation using Memcached. This implementation supports tags
cache/namespaced-cache               0.1.3              A decorator that makes your cache support namespaces
cache/psr-6-doctrine-bridge          dev-master ccfdf02 PSR-6 Doctrine bridge
cache/session-handler                1.0.0              An implementation of PHP's SessionHandlerInterface that allows PSR-6
cache/tag-interop                    1.0.0              Framework interoperable interfaces for tags
cache/taggable-cache                 1.0.0              Add tag support to your PSR-6 cache implementation
```

bug still exists.
Steps to reproduce
```
git clone https://github.com/zhil/cache-bundle-bug
cd cache-bundle-bug
composer install
./app/console app:test
```

**Comment by Nyholm** [comment]
This patch is working on your example project. Can you try it in your real application? 

https://github.com/php-cache/cache-bundle/pull/82

**Comment by zhil** [comment]
@Nyholm works great, thanks!

**Comment by Nyholm** [comment]
Awesome =)

**Comment by zhil** [comment]
@Nyholm since this patch fix bug - is it possible to add new tag? something like 1.0.1 would allow to use minimum-stability=stable

**Comment by Nyholm** [comment]
Sure. I'll to it tomorrow. 

//Tobias Nyholm

Skickat på språng

**Comment by zhil** [comment]
@Nyholm thanks for update.
for some reasons I cant upgrade bundle to 1.0.1. Composer just ignore that version.

```
composer show cache/cache-bundle -a 
# show 1.0.1 as available

composer show cache/cache-bundle 1.0.1
# show weird error about   Package cache/cache-bundle not found  

composer show cache/cache-bundle 1.0.0
# show 1.0.0 info
```

Its, probably, bug in composer, but before filling bug report - can you please try to execute
```
composer show cache/cache-bundle 1.0.1
```
Does it work for you?

**Comment by zhil** [comment]
![image](https://user-images.githubusercontent.com/981783/29066364-a8d85e88-7c37-11e7-93a1-09992028cf7e.png)

**Comment by Nyholm** [comment]
Strange. But FYI: 

https://github.com/php-cache/cache-bundle/pull/84#issuecomment-320909406

**Comment by Nyholm** [comment]
There. I've published 1.0.2: https://packagist.org/packages/cache/cache-bundle

About you installation issues. Have you required symfony/var-dumper anywhere? The new version have a dependency on `symfony/var-dumper: ^3.3`
